pkg/persistence/disk_persistence_test.go

```diff
@@ -46,8 +47,40 @@ func TestDiskPersistence_Save(t *testing.T) {
 	}
 }
 
+func TestDiskPersistence_StoragePermission(t *testing.T) {
+	tempDir := "./data_storage"
+
+	err := os.Mkdir(tempDir, 000) // d---------
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		t.Fatalf("dir [%+v] was supposed to be created", tempDir)
+	}
+	defer os.RemoveAll(tempDir)
+
+	_, err = NewDiskHandle(tempDir)
+	if err == nil {
+		t.Fatalf("error was supposed to be returned")
+	}
+	
+	expectedReadErr := "cannot read from the storage directory"
+	if !strings.Contains(err.Error(), expectedReadErr) {
+		t.Fatalf("error on read was supposed to be returned")
+	}
```

Comment:
It can be simplified by creating expected error with `fmt.Errorf` and comparing later using `reflect.DeepEqual`

Reply:
Based on your comment I started refactoring using `reflect.DeepEqual`, but then I noticed that Go lint actually doesn't recommend using `DeepEqual` for errors comparing. Currently it underlines it yellow (other files as well where we use DeepEqual)  I found [this](https://github.com/golang/go/issues/16748) github issue and this [stackoverflow](https://stackoverflow.com/questions/39121172/how-to-compare-go-errors) discussion  and based on them refactored a bit and left comparing errors without `DeepEqual`. 
Additionally, the system error will contain the data storage folder which can be named differently and that would've required additional parsing.